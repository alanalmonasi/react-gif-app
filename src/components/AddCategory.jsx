import PropTypes from 'prop-types'
import { useState } from "react"


export const AddCategory = ({ onNewValue }) => {

   const [inputValue, setInputValue] = useState('');

   const onInputChange = (e) => {
      setInputValue(e.target.value);
   }
   
   const onSubmit = (e) => {
      e.preventDefault();

      if(inputValue.length <= 1) return; 
      
      setInputValue('');

      onNewValue(inputValue.trim().toLocaleUpperCase());
   }

   return (
      <form onSubmit={onSubmit} aria-label="form">

         <input 
            type="text"
            placeholder="Buscar gif"
            value={inputValue}
            onChange={onInputChange}
         />
      </form>
   )
}

AddCategory.propTypes = {
   onNewValue: PropTypes.func.isRequired,
}