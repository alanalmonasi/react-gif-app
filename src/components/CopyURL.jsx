import { useState } from "react";

export const CopyURL = ({ url }) => {
   const [copied, setCopied] = useState(false);
   const link = url.split('?')

   const copy = () => { 
      const el = document.createElement('input');
      el.value = link[0];
      document.body.appendChild(el);
      el.select();
      document.execCommand('copy');
      document.body.removeChild(el);
      setCopied(true);
   }
   
   return (
      <div className="copy-link">
         <button onClick={copy}>{!copied ? "Copy link" : "Copied!"}</button>
      </div>
   );
}


