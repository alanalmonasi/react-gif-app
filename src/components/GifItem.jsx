import PropTypes from 'prop-types'

import React from 'react'
import { CopyURL } from './CopyURL'

export const GifItem = ({ title, url }) => {
  return (
      <div className="card">
         <img src={url} alt={title} />
         <p>{title}</p>
         <CopyURL url={url}/>
      </div>
   )
}

GifItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired
}
