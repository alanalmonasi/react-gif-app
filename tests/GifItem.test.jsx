import { render, screen } from "@testing-library/react"
import { GifItem } from "../src/components/GifItem"

describe('Testing the snapshot', () => { 
   
   const title = 'Simpons'
   const url = 'https://thesimpson.com/'

   test('should be the same as expected', () => { 
      
      const { container } = render(<GifItem title={title} url={url}/>);

      expect(container).toMatchSnapshot();

   });

   test('should show url and alt given', () => { 

      render(<GifItem title={title} url={url}/>);

      // expect(screen.getByRole('img').src).toBe(url);
      // expect(screen.getByRole('img').alt).toBe(title);
      const { src, alt } = screen.getByRole('img');

      expect(src).toBe(url);
      expect(alt).toBe(title);

   });

   test('should show component title', () => { 
   
      render(<GifItem title={title} url={url}/>);
      expect(screen.getByText(title)).toBeTruthy();

   })
})