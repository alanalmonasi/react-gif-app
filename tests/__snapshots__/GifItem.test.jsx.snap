// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the snapshot should be the same as expected 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Simpons"
      src="https://thesimpson.com/"
    />
    <p>
      Simpons
    </p>
  </div>
</div>
`;
